quantum/pointing_device/pointing_device_auto_mouse.c

```diff
@@ -31,63 +31,75 @@ static bool is_mouse_record(uint16_t keycode, keyrecord_t* record) {
+        local_auto_mouse.config.enabled = state;
         // reset settings on state change
-        auto_mouse.status.mouse_key_tracker = 0;
-        auto_mouse.timer.active             = 0;
-        auto_mouse.timer.delay              = 0;
-        auto_mouse.status.layer_toggled     = false;
+        local_auto_mouse.status.mouse_key_tracker = 0;
+        local_auto_mouse.timer.active             = 0;
+        local_auto_mouse.timer.delay              = 0;
+        local_auto_mouse.status.layer_toggled     = false;
     }
 }
 
 /* set which layer will act as the mouse layer */
 void set_auto_mouse_layer(uint8_t layer) {
-    auto_mouse.config.layer = layer;
+    local_auto_mouse.config.layer = layer;
 }
 
 /* toggle mouse layer setting (disabling automouse layer changes) */
 void toggle_mouse_layer(void) {
-    auto_mouse.status.layer_toggled ^= 1;
+    local_auto_mouse.status.layer_toggled ^= 1;
 }
 
 /* Allow for accessing toggle mouse bool value */
 bool get_toggle_mouse_state(void) {
-    return auto_mouse.status.layer_toggled;
+    return local_auto_mouse.status.layer_toggled;
 }
 
 /* Update the auto mouse if there is mouse motion on the base layer */
 void pointing_device_task_auto_mouse(report_mouse_t mouse_report) {
     // skip if disabled or layer not set
-    if (!auto_mouse.config.enabled) return;
+    if (!local_auto_mouse.config.enabled) return;
         // skip if oneshot mouse layer is active
 #    ifndef NO_ACTION_ONESHOT
-    if (get_oneshot_layer() == auto_mouse.config.layer) return;
+    if (get_oneshot_layer() == local_auto_mouse.config.layer) return;
 #    endif
     // Check for mouse movement and update
     if ((mouse_report.x != 0 || mouse_report.y != 0)) {
-        auto_mouse.timer.active = timer_read();
-        if (!layer_state_is(auto_mouse.config.layer) && timer_elapsed(auto_mouse.timer.delay) > AUTO_MOUSE_DELAY) {
-            layer_on(auto_mouse.config.layer);
-            auto_mouse.timer.delay = 0;
+        local_auto_mouse.timer.active = timer_read();
+        if (!layer_state_is(local_auto_mouse.config.layer) && timer_elapsed(local_auto_mouse.timer.delay) > AUTO_MOUSE_DELAY) {
+            layer_on(local_auto_mouse.config.layer);
+            local_auto_mouse.timer.delay = 0;
         }
-    } else if (timer_elapsed(auto_mouse.timer.active) > AUTO_MOUSE_TIME && layer_state_is(auto_mouse.config.layer) && !auto_mouse.status.mouse_key_tracker && !auto_mouse.status.layer_toggled) {
-        layer_off(auto_mouse.config.layer);
+    } else if (timer_elapsed(local_auto_mouse.timer.active) > AUTO_MOUSE_TIME && layer_state_is(local_auto_mouse.config.layer) && !local_auto_mouse.status.mouse_key_tracker && !local_auto_mouse.status.layer_toggled) {
+        layer_off(local_auto_mouse.config.layer);
     }
 }
 
 /* Reset timer and mouse button bool if mouse button is pressed */
-void process_auto_mouse(uint16_t keycode, keyrecord_t* record) {
+static void process_auto_mouse(uint16_t keycode, keyrecord_t* record) {
```

Comment:
Shouldn't  static here be removed?

Reply:
Yeah, this shouldn't be static, since you ***definitely*** need this to be accessed outside of this file.

Reply:
Yeah this is correct was static in a previous version before the PR but now shouldn't be anymore.